tests/race_test.go

```diff
@@ -90,3 +96,63 @@ func TestRaceBetweenTxpoolAddAndCommitNewWork(t *testing.T) {
 		t.Fatal(err)
 	}
 }
+
+// TestRaceAsMessageWithAccountPickerForFeePayer tests calling AsMessageWithAccountPicker of a fee delegated transaction
+// where a fee payer may be inserted wrongly due to concurrent issue.
+func TestRaceAsMessageWithAccountPickerForFeePayer(t *testing.T) {
+	if testing.Verbose() {
+		enableLog() // Change verbosity level in the function if needed
+	}
+
+	// Configure and generate a sample block chain
+	var (
+		gendb = database.NewMemoryDBManager()
+
+		// create a sender and a feepayer
+		from, _     = createAnonymousAccount("a5c9a50938a089618167c9d67dbebc0deaffc3c76ddc6b40c2777ae594389999")
+		feePayer, _ = createAnonymousAccount("ed580f5bd71a2ee4dae5cb43e331b7d0318596e561e6add7844271ed94156b20")
+
+		funds = new(big.Int).Mul(big.NewInt(1e16), big.NewInt(params.KLAY))
+		gspec = &blockchain.Genesis{
+			Config: params.TestChainConfig,
+			Alloc: blockchain.GenesisAlloc{
+				from.GetAddr():     {Balance: funds},
+				feePayer.GetAddr(): {Balance: funds},
+			},
+		}
+		genesis = gspec.MustCommit(gendb)
+		signer  = types.NewEIP155Signer(gspec.Config.ChainID)
+	)
+
+	iterNum := 10000
+	errCh := make(chan error, 2*iterNum)
+
+	for i := 0; i < iterNum; i++ {
+		tx, _ := genFeeDelegatedChainDataAnchoring(t, signer, from, nil, feePayer, big.NewInt(1234))
+		for i := 0; i < 2; i++ {
```

Comment:
I'm just curious that why you only make two goroutines per one iteration.

Reply:
I assumed that there are two goroutines which access `AsMessageWithAccountPicker` method, where a goroutine is for main and the other one is for prefetcher.